Add TT script enforcement tests using HTMLElement.innerText and Node.…

fred-wang opened the pull request:
…textContent

Setters for `HTMLElement.innerText` and `Node.textContent` can be retrieve via `HTMLElement.prototype.__lookupSetter__('innerText')` and `Node.prototype.__lookupSetter__('textContent')` respectively. New tests are added to make sure they allow to set the script source of a `<script>` element (without requiring to pass a `TrustedScript`) but that execution of such a script with modified source is blocked.

lukewarlow commented on trusted-types/script-enforcement-004.html:
```diff
@@ -79,6 +79,17 @@
+      container.appendChild(script);
+    });
+    assert_array_equals(messages, ["SET", "APPEND", "CREATE_SCRIPT", "HTMLScriptElement text", "RUN_TRUSTED_SCRIPT"]);
+  }, "Default policy's calls when setting script source via Node.prototype.__lookupSetter__('textContent').");
```
Slightly hard to understand this description the possessive followed by plural is tripping me up.

lukewarlow commented:
Btw is lookupsetter the only way to do this? Given it's deprecated should we use something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor instead. So modern engines can pass these tests without needed unrelated legacy stuff?

lukewarlow commented:
I'm just a little conscious of the fact both ladybird and servo are mid TT implementation so it would be nice to avoid obsolete stuff where possible. But perhaps that ship has sailed.

fred-wang committed:
Add TT script enforcement tests using HTMLElement.innerText and Node.textContent

Setters for `HTMLElement.innerText` and `Node.textContent` can be retrieve via
`Object.getOwnPropertyDescriptor()`. New tests are added to make sure they
allow to set the script source of a `<script>` element (without requiring to
pass a `TrustedScript`) but that execution of such a script with modified
source is blocked.

fred-wang commented:
> I'm just a little conscious of the fact both ladybird and servo are mid TT implementation so it would be nice to avoid obsolete stuff where possible. But perhaps that ship has sailed.

Yeah that makes sense. I've rewritten the test with `Object.getOwnPropertyDescriptor`. I think we don't really care how we retrieve the setters anyway, the important part is about how TT implementations will behave when they are used on scripts. So let's pick the standard API.

fred-wang committed:
fix the test desc again
